AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -162,67 +114,63 @@ - (void)setParent:(id<ASLayoutable>)parent
+
 - (void)setChild:(id<ASLayoutable>)child
 {
-  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");;
+  ASDisplayNodeAssert(_childrenArray.count < 2, @"This layout spec does not support more than one child. Use the setChildren: or the setChild:AtIndex: API");
+  
   if (child) {
     id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
     if (finalLayoutable) {
-      _children[0] = finalLayoutable;
+      _childrenArray[0] = finalLayoutable;
       [self propagateUpLayoutable:finalLayoutable];
     }
   } else {
-    _children.erase(0);
+    if (_childrenArray.count) {
+      [_childrenArray removeObjectAtIndex:0];
+    }
   }
 }
 
-- (void)setChild:(id<ASLayoutable>)child forIndex:(NSUInteger)index
+- (id<ASLayoutable>)child
 {
-  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
-  if (child) {
-    id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
-    _children[index] = finalLayoutable;
-  } else {
-    _children.erase(index);
+  ASDisplayNodeAssert(_childrenArray.count < 2, @"This layout spec does not support more than one child. Use the setChildren: or the setChild:AtIndex: API");
+  
+  if (_childrenArray.count) {
+    return _childrenArray[0];
   }
-  // TODO: Should we propagate up the layoutable at it could happen that multiple children will propagated up their
-  //       layout options and one child will overwrite values from another child
-  // [self propagateUpLayoutable:finalLayoutable];
+  
+  return nil;
 }
 
+#pragma mark - Children
+
 - (void)setChildren:(NSArray<id<ASLayoutable>> *)children
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
   
-  _children.clear();
+  [_childrenArray removeAllObjects];
+  
   NSUInteger i = 0;
   for (id<ASLayoutable> child in children) {
-    _children[i] = [self layoutableToAddFromLayoutable:child];
+    _childrenArray[i] = [self layoutableToAddFromLayoutable:child];
     i += 1;
   }
 }
 
-- (id<ASLayoutable>)childForIndex:(NSUInteger)index
+- (NSArray *)children
 {
-  if (index < _children.size()) {
-    return _children[index];
-  }
-  return nil;
+  return [_childrenArray copy];
 }
 
-- (id<ASLayoutable>)child
-{
-  return _children[0];
-}
+#pragma mark - NSFastEnumeration
 
-- (NSArray *)children
+- (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state objects:(id __unsafe_unretained _Nullable [_Nonnull])buffer count:(NSUInteger)len
 {
-  std::vector<ASLayout *> children;
-  for (ASChildMap::iterator it = _children.begin(); it != _children.end(); ++it ) {
-    children.push_back(it->second);
-  }
-  
-  return [NSArray arrayWithObjects:&children[0] count:children.size()];
+  return [_childrenArray countByEnumeratingWithState:state objects:buffer count:len];
```

Comment:
`NSMutableArray` internally should throw an assertion if something is mutating while iterating